### PR TITLE
Replace setup.py + requirements.txt with pyproject.toml

### LINKED ISSUE #497: Requirements.txt and Pipfile are not in sync
**Describe the bug**
The requirements.txt and Pipfile have different indications

**To Reproduce**
pipenv update --dev
pipenv run python setup.py test

**Expected behavior**
test suite runs

**Additional context**
See failing test in #494

**Proposed resolution**
Remove `requirements.txt`

@mayaCostantini
@harshad16
wdyt ?

### PR DESCRIPTION
**Pull request by VannTen**
- Use micropipenv for the build process to handle the Pipfile.lock
fix #497

### CODE REVIEW

**Review by harshad16 — COMMENTED** [review]
/lgtm


The image build and package release by testing works.
thanks for this :100: 
Though it requires to rebase as few package update pr were merged.

File: Pipfile [hunk 1]
```diff
@@ -13,6 +13,7 @@ thoth-analyzer = "*"
 prometheus-client = "*"
 pip = "*"
 toml = "*"
+package-extract = {path = "."}
```
Comment (KPostOffice) [inline Pipfile:16]: Will this break the image? I know installing the local package is sometimes necessary for the packages CLIs to be in `$PATH`

**Comment by VannTen** [comment]
Needs approval o7

**Comment by VannTen** [comment]
@harshad16 or @KPostOffice : some approval pls ? :)

File: Dockerfile [hunk 2]
```diff
@@ -22,9 +22,11 @@ RUN dnf update -y --setopt='tsflags=nodocs' && \
     dnf install -y binutils
 
 COPY ./ /tmp/package-extract
-RUN cd /tmp/package-extract && \
-    make all && \
-    rm -rf /tmp/package-extract
+RUN pip install micropipenv \
+        && cd /tmp/package-extract \
+        && micropipenv install \
+        && pip install --no-deps . \
```
Comment (harshad16) [inline Dockerfile:28]: why is this required ?

**Review by harshad16 — COMMENTED** [review]
few questions:
shall we remove the makefile too with this PR ?

File: Dockerfile (hunk 2 above)
Comment (VannTen) [reply Dockerfile:28]: Indeed we should.
You mean the pip part ?
Micropipenv can't currently handle 'build + install' process (`micropipenv install .` result in a never-ending loop) so we just use it to install the exact deps  (specified in Pipfile.lock) and tell pip to build and install the Python package in the current dir.

Comment (harshad16) [reply Dockerfile:28]: acknowledge thanks.

**Comment by codificat** [comment]
/lgtm
thanks!
